deepchem/metrics/__init__.py

```diff
@@ -161,6 +161,59 @@ def kappa_score(y_true, y_pred):
   return kappa
 
 
+def bedroc_score(y_true, y_pred, alpha=20.0):
```

Comment:
Can we use the Bedroc code already in rdkit?
http://www.rdkit.org/Python_Docs/rdkit.ML.Scoring.Scoring-pysrc.html#CalcBEDROC

{code}
from rdkit.ML.Scoring.Scoring import CalcBEDROC
{code}

{code}
y_true = np.flatten(y_true)
y_pred = np.flatten(y_pred[:,1])
scores = list(zip(y_true, y_pred))
return CalcBEDROC(scores, 0, alpha)
{code}

Reply:
The source for CalcBEDROC is a little confusing.

From the paper, what I gauged the idea should be is:
1. Sort the predicted scores/labels in increasing order and get the sorted indices.
2. Get ranks for active molecules by selecting the active molecule numbers from the sorted indices
3. Compute the expression for RIE.

What CalcBEDROC does:
1. We specify scores (which according to their unittest is a list of single element lists)
2. They directly check on the scores and compute number of actives and the RIE from that.

From the above code, the confusion stems from the fact that, if we supply scores as a list of (y_true, y_pred) pairs, the calcBEDROC never looks at the predictions to compute the RIE, which is weird.

If I use the reverse (ie (y_pred, y_true)) pairs, then the number of actives are directly computed from the y_pred values, which is again weird, because of presence of False Positives and Negatives in the prediction.

Reply:
Ah I think we need to sort before calling CalcBedroc.

```python
from rdkit.ML.Scoring.Scoring import CalcBEDROC
y_true = np.flatten(y_true)
y_pred = np.flatten(y_pred[:,1])
scores = list(zip(y_true, y_pred))
scores = sorted(scores(scores, key=lambda x: x[1], reverse=True) # High is good
return CalcBEDROC(scores, 0, alpha)
```

If we sort first the RDKit code seems to be doing the algorithm laid out by you.

Reply:
If I were to rewrite the docstring for CalcBedRoc after reading the source code.

```python
def CalcBEDROC(scores, col, alpha):
    """BEDROC original defined here: 
        Truchon, J. & Bayly, C.I. 
        Evaluating Virtual Screening Methods: Good and Bad Metric for the "Early Recognition" 
        Problem. J. Chem. Inf. Model. 47, 488-508 (2007). 

        scores: 2d sequence
             scores are a 2d sequence with the 0th index representing sample
             scores[sample_id] = vector of sample data
        col: int
             Index of sample data which reflects true label of a sample
             scores[sample_id][col] = True iff that sample is active
        alpha: float
             hyper parameter from the initial paper for how much to enrich the top

        
    """
```

Reply:
Ah, I just checked their Github page 
https://github.com/rdkit/rdkit/blob/master/rdkit/ML/Scoring/Scoring.py

It says:
```scores: ordered list with descending similarity containing active/inactive information```